lib/server/db/seeds/connection.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Utility for adding configurable Connections seeds to the
+ * database.
+ *
+ * @module ConnectionSeed
+ */
+var User = require("../model/User");
+var Connection = require("../model/Connection");
+
+/**
+ * @typedef {Object} ConnectionSeedConfig
+ * @property {String} ConnectionSeedConfig.user - The from-User
+ * @property {String[]} ConnectionSeedConfig.friends - The to-Users
+ */
+
+/**
+ * Create connections between a user and a variable number of friends and
+ * returns a then-able Promise, passing the connections array to the next "then".
+ *
+ * @private
+ * @function createConnections
+ * @param {ConnectionSeedConfig} config - The name of the from-User
+ * @returns {Promise}
+ */
+function createConnections(config) {
+    return User
+        .findOne({
+            username: config.user
+        })
+        .then(function (user) {
+            return User
+            // find all the input users
+                .find({
+                    $or: config.friends.map(function (friend) {
+                        return {
+                            username: friend
+                        };
+                    })
+                })
+                // bundle up the user and friends to pass as data
+                .then(function (friends) {
+                    return {
+                        user: user,
+                        friends: friends
+                    };
+                });
+        })
+        .then(function (data) {
+            return data.friends.map(function (friend) {
+                return new Connection({
+                    user: data.user._id,
+                    friend: friend._id
+                });
+            });
+        })
+        .catch(console.error);
+}
+
+/**
+ * Saves an array of Connection instances
+ *
+ * @private
+ * @function saveConnections
+ * @param {Connection[]} connections - An array of Connections
+ * @returns {undefined}
+ */
+function saveConnections(connections) {
+    connections.forEach(function (connection) {
+        connection
+            .save()
+            .then(function(result, number) {
```

Comment:
Can you explain what result and number are in this function? Give me an example of what this would log out to the console if you just successfully added me as one of your connections.